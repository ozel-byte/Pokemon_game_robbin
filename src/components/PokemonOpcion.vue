<template>
  <div class="options-container">
    <ul>
      <li  :class="prueba[index]" @click="$emit('selectionPokemon',pokemon.id )" v-for="(pokemon, index) in pokemons" :key="pokemon.id">
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
  },
  data(){
    return{
      prueba: [
        "fondo1",
        "fondo2",
        "fondo3",
        "fondo4"
      ],
      ArrayColors: [
        {
          textColor: "#334859",
          fondoCOlor: "#A7B7BB"
        },
        {
          textColor: "#401f46",
          fondoCOlor: "#c9becc"
        },
        {
          textColor: "#a76932",
          fondoCOlor: "#faeadd"
        },
        {
          textColor:  "#4f5925",
          fondoCOlor: "#b7b59c"
        }
      ]
    }
  }
};
</script>

<style scope>
ul {
  list-style-type: none;
}
.fondo1{
  background-color: #A7B7BB;
  color: #334859;
  border:1px solid #A7B7BB;
}
.fondo2{
background-color:#c9becc ;
  color: #401f46;
  border: 1px solid #c9becc;
}
.fondo3{
background-color:#faeadd;
  color: #a76932;
  border: 1px solid #faeadd;
}
.fondo4{
background-color: #b7b59c;
  color:#4f5925 ;
  border: 1px solid #b7b59c;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
  padding: 10px;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  color: #334859;
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>