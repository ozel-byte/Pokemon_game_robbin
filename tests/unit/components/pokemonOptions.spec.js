import {shallowMount} from '@vue/test-utils'
import PokemonOpcion from '@/components/PokemonOpcion';

describe( 'pokemonOptions', () => {

    let wrapper;

    beforeEach(()=>{
        wrapper = shallowMount(PokemonOpcion, {
            props:{
                pokemons: [
                    {name: 'bulbasaur', id: 1},
                    {name: 'ivysaur', id:   2},
                    {name: 'venusaur', id:  3},
                    {name: 'charmander', id: 4}
                ]
            }
        })
    })
    test( 'debe ser match', () => {
       console.log(wrapper.html());

       expect(wrapper.html()).toMatchSnapshot();
    });

    test( ' debe mostrar  las 4 opciones correctamente', () => {
        const app = wrapper.findAll("li");
        expect(app.length).toBe(4);
        const [p1,p2,p3,p4] = app;
        expect(p1.text()).toBe("bulbasaur");
        expect(p2.text()).toBe("ivysaur");
        expect(p3.text()).toBe("venusaur");
        expect(p4.text()).toBe("charmander");
    })

    test( 'debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
        const [li1,li2,li3,li4] = wrapper.findAll("li");
        console.log("Buenas");
       
        li1.trigger("click");
        li2.trigger("click");
        li3.trigger("click");
        li4.trigger("click");
        console.log(wrapper.emitted('selectionPokemon'));
        console.log(wrapper.emitted('selectionPokemon')[0]);
         expect(wrapper.emitted('selectionPokemon').length).toBe(4);
        expect(wrapper.emitted('selectionPokemon')[0]).toEqual([1]);
        expect(wrapper.emitted('selectionPokemon')[1]).toEqual([2]);
        expect(wrapper.emitted('selectionPokemon')[2]).toEqual([3]);
        expect(wrapper.emitted('selectionPokemon')[3]).toEqual([4]);
    })
});