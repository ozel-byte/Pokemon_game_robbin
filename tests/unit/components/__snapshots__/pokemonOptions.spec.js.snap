// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonOptions debe ser match 1`] = `
<div class="options-container">
  <ul>
    <li class="fondo1">bulbasaur</li>
    <li class="fondo2">ivysaur</li>
    <li class="fondo3">venusaur</li>
    <li class="fondo4">charmander</li>
  </ul>
</div>
`;
